src/api/ThreadedComments.js

```diff
@@ -0,0 +1,340 @@
+/**
+ * @flow
+ * @file Helper for the box threadedComments API
+ * @author Box
+ */
+
+import type { ThreadedCommentPermission, ThreadedCommentStatus } from '../common/types/threadedComments';
+import MarkerBasedApi from './MarkerBasedAPI';
+import {
+    PERMISSION_CAN_COMMENT,
+    PERMISSION_CAN_DELETE,
+    PERMISSION_CAN_EDIT,
+    ERROR_CODE_CREATE_COMMENT,
+    ERROR_CODE_UPDATE_COMMENT,
+    ERROR_CODE_DELETE_COMMENT,
+    ERROR_CODE_FETCH_COMMENTS,
+    PERMISSION_CAN_RESOLVE,
+    ERROR_CODE_FETCH_REPLIES,
+    ERROR_CODE_CREATE_REPLY,
+} from '../constants';
+import type { ElementsXhrError, ElementsErrorCallback } from '../common/types/api';
+import type { BoxItem, BoxItemPermission } from '../common/types/core';
+
+class ThreadedComments extends MarkerBasedApi {
+    /**
+     * API URL for comments
+     *
+     * @return {string} base url for comments
+     */
+    getUrl(): string {
+        return `${this.getBaseApiUrl()}/undoc/comments`;
+    }
+
+    /**
+     * API URL for specific comment
+     *
+     * @param {string} [commentId]
+     * @return {string} base url for specific comment
+     */
+    getUrlForId(commentId: string): string {
+        return `${this.getUrl()}/${commentId}`;
+    }
+
+    /**
+     * API URL for specific comment
+     *
+     * @param {string} [commentId]
+     * @return {string}  base url for specific comment replies
+     */
+    getUrlWithRepliesForId(commentId: string): string {
+        return `${this.getUrlForId(commentId)}/replies`;
+    }
+
+    /**
+     * API for creating a comment on a file
+     *
+     * @param {BoxItem} file - File object for which we are creating a comment
+     * @param {string} message - Comment message
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @return {void}
+     */
+    create({
+        file,
+        message,
+        successCallback,
+        errorCallback,
+    }: {
+        errorCallback: ElementsErrorCallback,
+        file: BoxItem,
+        message?: string,
+        successCallback: Function,
+    }): void {
+        this.errorCode = ERROR_CODE_CREATE_COMMENT;
+        const { id = '', permissions, type } = file;
```

Comment:
Can `id` be undefined? It seems like that should throw an error.

Reply:
You're right 👍